scripts/generatePortVersionsDb.py

```diff
@@ -0,0 +1,76 @@
+import os
+import os.path
+import sys
+import subprocess
+import json
+import time
+
+from subprocess import CalledProcessError
+from json.decoder import JSONDecodeError
+from pathlib import Path
+
+def make_dir(dir_path):
+    if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+
+
+def get_current_git_ref():
+    try: 
+        output = subprocess.run(args=['git.exe', 'rev-parse', '--verify', 'HEAD'], capture_output=True)
+        if output.returncode == 0:
+            return output.stdout.decode('utf-8').strip()
+        return ''
+    except CalledProcessError as err:
+        print("Failed to get git ref")

```

Comment:
```suggestion
    output = subprocess.run(['git', '-C', SCRIPT_DIRECTORY, 'rev-parse', '--verify', 'HEAD'], capture_output=True, encoding='utf-8')
    if output.returncode == 0:
        return output.stdout.strip()
    print('Failed to get current git ref:', output.stderr.strip(), file=sys.stderr)
    return None
```

Few things:

* `subprocess.run` doesn't raise a `CalledProcessError` unless you pass `check=True`
* Correctly outputting to stderr, and outputting the actual error message
* Use `git -C <path>` to make sure that we're using the right git repository
* (minor) returning None if there's an error is better than empty string
* (minor) adding the encoding in `run` is easier.